infra/Pos/KnownPeers.hs

```diff
@@ -28,3 +31,10 @@ instance {-# OVERLAPPABLE #-}
     ( Monad m, MonadTrans f, MonadFormatPeers m ) =>
         MonadFormatPeers (f m) where
   formatKnownPeers k = lift (formatKnownPeers k)
+
+class HasOutboundQ m where
+  type Pack m :: * -> *
```

Comment:
Alas I've got into dependency cycles hell, thus the remedial associated type 😞 

Hopefully though I can simply kill all this code as you guys pointed out we can do everything inside `listenerConv` anyway!